include/kernel/thread.h

```diff
@@ -307,5 +306,7 @@ void awaken(void);
 /* Place the thread bound to this scheduling context in the release queue
  * of periodic threads waiting for budget recharge */
 void postpone(sched_context_t *sc);
+#else
+#define updateTimestamp()
```

Comment:
Can you leave a comment that we have this as a macro define to nothing because this avoids having to update the proofs. Otherwise this will be another place in the code that looks a bit odd when having a pure C view.

Reply:
@lsf37: can't we make this `#define updateTimestamp() ((void)(0))` (or `do {} while false`) because the proofs can handle this as a no-op? Then we'd have statement here that requires/consume a `;` and looks like a function. It's also what we are doing with `printf()` in https://github.com/seL4/seL4/blob/812c0c2a395e3179e7119d7b9df4f153a19132a2/include/machine/io.h#L110-L117

Reply:
I still don't know if semicolons need manual proof updates for non-MCS or not, so I went with @kent-mcleod's suggestion and stuffed everything in the `config_ternary`.